pkg/compactor/block_upload.go

```diff
@@ -529,3 +634,43 @@ func (c *MultitenantCompactor) loadValidation(ctx context.Context, userBkt objst
 
 	return v, nil
 }
+
+func (c *MultitenantCompactor) uploadValidationWithError(ctx context.Context, blockID ulid.ULID,
```

Comment:
This is very similar to `uploadMeta`. Can we have single `marshalAndUploadToBucket` function that extracts common code (marshalling, uploading), similar to `marshalAndUploadJSON` from the test code. (Test code could reuse it too)